include/swift/SIL/SILInstruction.h

```diff
@@ -7913,6 +7924,15 @@ class FullApplySite : public ApplySite {
     return getArguments().slice(getNumIndirectSILResults());
   }
 
+  /// Returns true if \p oper is the callee operand and not an argument operand.
+  bool isCalleeOperand(const Operand &oper) const {
+    return oper.getOperandNumber() < getOperandIndexOfFirstArgument();
+  }
+
+  bool isIndirectResultOperand(const Operand &oper) const {
+    return oper.getOperandNumber() < getNumIndirectSILResults();
```

Comment:
This isn't right is it? Shouldn't it be `getCalleeArgIndex(oper) < getNumIndirectSILResults`?

Reply:
As we discussed via chat:

```
bool isIndirectResultOperand(const Operand &oper) const {
  return getCalleeArgIndex(oper) < getSILArgIndexOfFirstParam();
}
```